File: js/src/Blockchain/Blockchain.Abstraction/Models/ReceiptModels/TransactionResponse.ts
import { TransactionStatus } from "../TransacitonModels/TransactionStatus";

export interface TransactionResponse {
    amount?: string;
    asset?: string;
    decimals: number;
    networkName: string;
    transactionHash: string;
    confirmations: number;
    timestamp: Date;
    feeAmount: string;
    feeAsset: string;
    feeDecimals: number;
    status: TransactionStatus;
  }

File: csharp/src/Workflow.Abstractions/Models/TransactionResponse.cs
using System.Numerics;
using Train.Solver.Common.Enums;

namespace Train.Solver.Workflow.Abstractions.Models;

public class TransactionResponse
{
    public required string NetworkName { get; set; } = null!;

    public required string TransactionHash { get; set; } = null!;

    public required int Confirmations { get; set; }

    public required DateTimeOffset Timestamp { get; set; }

    public required BigInteger FeeAmount { get; set; }

    public required string FeeAsset { get; set; }

    public required int FeeDecimals { get; set; }

    public required TransactionStatus Status { get; set; }
}
